opencog/query/PatternUtils.cc

```diff
@@ -117,21 +116,18 @@ void get_connected_components(
 			size_t nc = components.size();
 			for (size_t i = 0; i<nc; i++)
 			{
-				std::set<Handle> cur_vars(component_vars[i]);
```

Comment:
To "fool" the compiler into constness one can decorate the members with the mutable keyword. I don't think it creates mess on the compiler's end. From the user's perspective I guess it means that we consider those mutable members will have no external side effects.

Regarding Atom, the tricky thing is that it inherits std::enable_shared_from_this, and then getHandle calls shared_from_this(), now I checked and there are 2 versions of it (in shared_ptr.h)

```
      shared_ptr<_Tp>
      shared_from_this()
      { return shared_ptr<_Tp>(this->_M_weak_this); }

      shared_ptr<const _Tp>
      shared_from_this() const
      { return shared_ptr<const _Tp>(this->_M_weak_this); }
```
so apparently our code is calling the non const version. I'm not sure yet how to deal with that...